pkg/server/plugin/notifier/k8sbundle/k8sbundle.go

```diff
@@ -144,10 +147,15 @@ func (p *Plugin) Configure(ctx context.Context, req *configv1.ConfigureRequest)
 		}
 		setDefaultValues(&config.Clusters[i])
 	}
+	p.setConfig(config)
 
-	if err = p.setConfig(config); err != nil {
-		return nil, status.Errorf(codes.Internal, "unable to set configuration: %v", err)
+	clients, err := p.hooks.newKubeClients(config)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unable to create new kubeClients: %v", err)
 	}
+	p.setClients(clients)
+
+	p.startInformers(config, clients)
```

Comment:
yes, good point. do you think its worth keep the refactor or revert. I didn't like recreating the clients each time, but then again it doesn't run very often so i might be over optimizing.

Reply:
We have to create the new clients anyway to start the informers so I think what you have is fine, just needs a slight reshuffle so that we we only replace the existing configuration and clients once we've successfully parsed the new config, created the new clients, and started the informers. In other words, don't have a separate setClients and setConfig but rather, after the clients have been used to successfully start the informers, the config and clients get replaced at once. startInformers could do this (start informing, wait for the initial sync, take lock, replace config, clients, and stopCh).

Reply:
updated to address the concerns